### ISSUE TITLE
floor_div runtime error for i64, u32 and u64

### ISSUE THREAD

**Issue by TitouanReal** [pola-rs/polars#17238]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```rust
use polars::df;
use polars::prelude::*;

fn main() {
    let df = df![
        "seconds" => [0u64],
    ]
    .unwrap();

    let new_df = df
        .lazy()
        .with_column(col("seconds").floor_div(lit(1u64)))
        .collect()
        .unwrap();

    println!("{new_df:?}");
}

```

### Log output

```shell
run StackExec
thread 'main' panicked at /home/titouan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-core-0.41.2/src/series/mod.rs:956:13:
implementation error, cannot get ref UInt64 from Int32
```

### Issue description

The example given above worked in version 0.39.2, but it is broken from 0.40.0 and onwards.

Variations of this error appear when using u64 (given example), u32, and i64, with the error becoming:
- implementation error, cannot get ref UInt64 from Int32,
- implementation error, cannot get ref UInt32 from Int32,
- implementation error, cannot get ref Int64 from Int32,

respectively.

### Expected behavior

In version 0.39.2, the example works and outputs :

shape: (1, 1)
┌─────────┐
│ seconds         │
│ ---                    │
│ u64                 │
╞═════════╡
│ 0                      │
└─────────┘

which is the expected output.

### Installed versions

```toml
[dependencies.polars]
version = "0.41"
default-features = false
features = ["fmt", "lazy", "round_series"]
```

**Comment by TitouanReal** [comment]
A solution to make this work is to use the following code instead:

```rust
use polars::df;
use polars::prelude::*;

fn main() {
    let df = df![
        "seconds" => [0u64],
    ]
    .unwrap();

    let new_df = df
        .lazy()
        .with_column((col("seconds") / lit(1u64)).floor().cast(DataType::UInt64))
        .collect()
        .unwrap();

    println!("{new_df:?}");
}
```

Note that this solution, while working in 0.41, panics in version 0.39.

The performance might be degraded compared to the code proposed in the issue description, but at least it doesn't panic, and it shows the expected behavior.

**Comment by TitouanReal** [comment]
I can confirm that this panic still occurs in 0.42.0 and 0.43.1.

**Comment by TitouanReal** [comment]
This is now fixed in 45.1. Closing.
